File: Problem392/Solution.go
package main

func isSubsequence(s string, t string) bool {
	if len(s) > len(t) {
		return false
	}
	i := 0
	j := 0
	n := len(s)
	m := len(t)
	for i < n {
		for j < m && t[j] != s[i] {
			j++
		}
		if j >= m {
			return false
		}
		i++
		j++
	}
	return true
}

File: Problem392/Solution.java

public class Solution
{
    public boolean isSubsequence(String s, String t)
    {
        if (s.length() > t.length())
        {
            return false;
        }
        int i = 0;
        int j = 0;
        while (i < s.length())
        {
            while (j < t.length() && t.charAt(j) != s.charAt(i))
            {
                j++;
            }
            if (j >= t.length())
            {
                return false;
            }
            i++;
            j++;
        }
        return true;
    }
}